File: clear-extjs-runtime-core/src/clear/transaction/BatchMember.java
/**
 * Copyright (c) 2009 Farata Systems  http://www.faratasystems.com
 *
 * Licensed under The MIT License
 * Re-distributions of files must retain the above copyright notice.
 *
 * @license http://www.opensource.org/licenses/mit-license.php The MIT License
 *
 */
package clear.transaction;
import java.io.Serializable;
import java.util.*;

public class BatchMember implements Serializable{
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	public String   methodName;
	public String   className;
	@SuppressWarnings("rawtypes")
	public List     parameters;
	public String autoSyncSubtopic;
	public Object   result;

	public String getClassName()	{
		return className;
	}
	
	public String getMethodName()	{
		return methodName;
	}

	
	@SuppressWarnings("rawtypes")
	public List getParameters()	{
		return parameters;
	}
	
	public String getAutoSyncSubtopic()	{
		return autoSyncSubtopic;
	}

	public Object getResult()	{
		return result;
	}
	public void setClassName(String s)	{
		className = s;
	}
	
	public void setMethodName(String s)	{
		methodName = s;
	}

	public void setAutoSyncSubtopic(String s)	{
		autoSyncSubtopic = s;
	}

	@SuppressWarnings("rawtypes")
	public void setParameters(List lst)	{
		parameters = lst;
	}

	public void setResult(Object obj)	{
		result = obj;
	}
	
	public final String __type__="batchMember";
	
}

File: com.farata.cleardatabuilder.extjs/resources/new_project/WebContent/clear/transaction/BatchMember.js
/*

This file is part of Clear-ExtJS for Ext JS 4

Copyright (c) 2011 Farata Systems LLC

Licensed under The MIT License
Re-distributions of files must retain the above copyright notice.

@license http://www.opensource.org/licenses/mit-license.php The MIT License

*/

/**
 * This class is used by {@link Clear.transaction.BatchService BatchService to facilitates sequential invoke of several remote methods without round-trips to the client. 
 * Request sent to the server and response from the server are both contain array of BatchMember instances.
 */
Ext.define('Clear.transaction.BatchMember', {
    /* Begin Definitions */
    alias: 'batchMember',
    
	className:'',
	methodName:'',
	parameters:[],
	autoSyncSubtopic:'',
	result:{},

    /* End Definitions */
    
    
    constructor: function(config) {
    	Ext.apply(this,config); 
    	this.toString = function() {
			var me = this;
			return [
				"clear.transaction.BatchMember {",
				" \nclassName: ", me.className,
				" \nmethodName: ", me.methodName,
				" \nparameters: ", me.parameters,
				" \nautoSyncSubtopic: ", me.autoSyncSubtopic,
				"\n}"
			].join(""); 
		};     
    }
});




